cpp/dolfinx/fem/utils.h

```diff
@@ -239,11 +239,34 @@ la::SparsityPattern create_sparsity_pattern(const Form<T, U>& a)
   return pattern;
 }
 
-/// Create an ElementDofLayout from a ufcx_dofmap
-ElementDofLayout create_element_dof_layout(const ufcx_dofmap& dofmap,
-                                           const mesh::CellType cell_type,
+/// Create an ElementDofLayout from a FiniteElement
+template <std::floating_point T>
+ElementDofLayout create_element_dof_layout(const fem::FiniteElement<T>& element,
                                            const std::vector<int>& parent_map
-                                           = {});
+                                           = {})
+{
+  // Create subdofmaps and compute offset
+  std::vector<int> offsets(1, 0);
+  std::vector<dolfinx::fem::ElementDofLayout> sub_doflayout;
+
+  for (int i = 0; i < element.num_sub_elements(); ++i)
```

Comment:
This code is quite hard to read. Could it be simplified?

Reply:
I've added some comments to explain the conditionals. The other options is to change it to this, but I don't like the additional code duplication it adds:

```c++
if (element.block_size() > 1)
  { 
    // blocked element
    std::shared_ptr<const fem::FiniteElement<T>> sub_e
        = element.sub_elements()[0];
    for (int i = 0; i < element.num_sub_elements(); ++i)
    {
      offsets.push_back(offsets.back() + 1);
      std::vector<int> parent_map_sub(sub_e->space_dimension());
      for (std::size_t j = 0; j < parent_map_sub.size(); ++j)
        parent_map_sub[j] = offsets[i] + element.block_size() * j;
      sub_doflayout.push_back(
          dolfinx::fem::create_element_dof_layout(*sub_e, parent_map_sub));
    }
  } else {
    // Mixed element or element with no sub-elements
    for (int i = 0; i < element.num_sub_elements(); ++i)
    {
      std::shared_ptr<const fem::FiniteElement<T>> sub_e
          = element.sub_elements()[i];
      offsets.push_back(offsets.back() + sub_e->space_dimension());
      std::vector<int> parent_map_sub(sub_e->space_dimension());
      for (std::size_t j = 0; j < parent_map_sub.size(); ++j)
        parent_map_sub[j] = offsets[i] + j;
      sub_doflayout.push_back(
          dolfinx::fem::create_element_dof_layout(*sub_e, parent_map_sub));
    }
  }
```